src/components/ChatBox.js

```diff
@@ -0,0 +1,240 @@
+			try {
+				const storedMessages = JSON.parse(localStorage.getItem(storageKey));
+
+				if (storedMessages && storedMessages.length > 0) {
+					setMessages(storedMessages);
+					const latestMessage = storedMessages[storedMessages.length - 1];
+					if (latestMessage?.timeStamp) {
+						setLatestMessageTimestamp(latestMessage.timeStamp);
+					}
+				}
+
+				const conversation = await pollConversation(
+					conversationId,
+					latestMessageTimestamp
+				);
+
+				if (
+					conversation &&
+					typeof conversation === 'object' &&
+					Array.isArray(conversation.messages)
+				) {
+					const updatedMessages = storedMessages
+						? [...storedMessages, ...conversation.messages]
+						: conversation.messages;
+
+					setMessages(updatedMessages);
+					localStorage.setItem(storageKey, JSON.stringify(updatedMessages));
+
+					const latestMessage = updatedMessages[updatedMessages.length - 1];
+					if (latestMessage?.timeStamp) {
+						setLatestMessageTimestamp(latestMessage.timeStamp);
+					}
+				} else if (!storedMessages) {
+					setErrorMessage('No messages found for this conversation.');
+				}
+			} catch (error) {
+				console.log(error);
+				setErrorMessage('Failed to fetch conversation messages.');
+			} finally {
+				setLoading(false);
+			}
+		};
+
+		fetchMessages();
+
+		const pollMessages = setInterval(async () => {
+			try {
+				if (latestMessageTimestamp) {
+					const newMessages = await pollConversation(
+						conversationId,
+						latestMessageTimestamp
+					);
+					if (
+						newMessages &&
+						Array.isArray(newMessages.messages) &&
+						newMessages.messages.length > 0
+					) {
+						setMessages((prevMessages) => {
+							const updatedMessages = [
+								...prevMessages,
+								...newMessages.messages,
+							];
+							localStorage.setItem(storageKey, JSON.stringify(updatedMessages));
+							return updatedMessages;
+						});
+
+						const latestMessage =
+							newMessages.messages[newMessages.messages.length - 1];
+						if (latestMessage?.timeStamp) {
+							setLatestMessageTimestamp(latestMessage.timeStamp);
+						}
+					}
+				}
+			} catch (error) {
+				console.error('Error polling for new messages:', error);
+			}
+		}, 1000);
+		return () => clearInterval(pollMessages);
+	}, [conversationId, latestMessageTimestamp, storageKey]);
+
+	useEffect(() => {
+		if (
+			messagesEndRef.current &&
+			typeof messagesEndRef.current.scrollIntoView === 'function'
+		) {
+			messagesEndRef.current.scrollIntoView({ behavior: 'smooth' });
+		}
+	}, [messages]);
+
+	const handleSendMessage = async () => {
+		if (!newMessage.trim()) return;
+
+		const postMessageRequest = {
+			conversationId,
+			contents: newMessage.trim(),
+		};
+
+		try {
+			setLoading(true);
+			const response = await postMessage(postMessageRequest);
+			if (response?.error) {
+				setErrorMessage(response.error);
+			} else {
+				const newMsg = {
+					contents: newMessage,
+					timeStamp: new Date().toISOString(),
```

Comment:
I think this might cause a duplication issue with the Logic/DB setting the message time stamp differently and then polling adding the new message to the list, We'll have to live test to know for sure, I don't think its worth changing yet.

Reply:
That's a fair argument, once we do live tests we can do the minor changes.